### ISSUE TITLE
Listing a long chat history makes qTox crash

### ISSUE THREAD

**Issue by ghost** [qTox/qTox#4134]
##### Brief Description

OS: Linux (Arch Linux)
qTox version: v1.7.1
Commit hash:  6f155d4db806a664d76457d42a20246ffbd5d1ac
toxcore:  0.1.5
Qt:  5.7.1
Hardware:  Generic x86 PC hardware
…
Reproducible: Always

##### Steps to reproduce

1.  Right-click in a conversation window that has a lot of history, a year or so.
2. Select "Load chat history..."
3. Go back as far as you think the chat history goes.

##### Observed Behavior

qTox freezes, and I can't do anything with it. The window starts updating, and I have to kill the process.

##### Expected Behavior
qTox should list the requested history, allowing me to save it down.

##### Additional Info
It sorta works (albeit slow) when I just want to list a couple months of history. The option to save your whole chat history without having to list it in the client could be a nice function.
----
[qtox.txt](https://github.com/qTox/qTox/files/752692/qtox.txt)

**Comment by tWido** [comment]
Option to export history to plaintext has been added. Is there anything more to be done?

**Comment by ghost** [comment]
Well, the issue obviously persists, but this makes it possible to export history while not having to list it all.

**Comment by Diadlo** [comment]
@Amitari Ok, I will reopen it

**Comment by tWido** [comment]
@Amitari @Diadlo Don't wanna be a pessimist, but loading 100K+ messages in graphical window will surely cause problems. Probably exceeding memory. 
One workaround could be paging i.e. count of messages loaded in window would be limited. Upon scrolling to bottom newer would be shown and old ones hidden.
Second way is adding range to history selection. So user could select messages in between dates and load messages in smaller batches.

**Comment by Diadlo** [comment]
@tWido Yes, I discussed it with @noavarice. We decided, that first workaround is good enough (load only visible messages)

**Comment by Diadlo** [comment]
But it's hard to implement with current architecture. We should start with creating `Message` model
